addon/-private/record-array-query.js

```diff
@@ -0,0 +1,41 @@
+export default class RecordArrayQuery {
+
+  constructor(store, type, params) {
+    this.store = store;
+    this.type = type;
+    this.params = params;
+
+    /**
+      The latest result of the query.
+
+      @public
+      @property value
+    */
+    this.value = null;
+  }
+
+  /**
+    Execute the query against the network.
+
+    @public
+    @method run
+  */
+  run() {
+    let promise;
+
+    if (this.value) {
+      promise = this.value.update();
+
+    } else {
+      promise = this.store.query(this.type, this.params)
+        .then(records => {
+          this.value = records;
+
+          return records;
+        });
```

Comment:
Does `this.value.update()` return records as well?

Reply:
`value` is a [AdapterPopulatedRecordArray](https://emberjs.com/api/ember-data/2.16/classes/DS.AdapterPopulatedRecordArray) so that would delegate to that class. Good question though, I can add a test.

Reply:
I think a good test would be if I render this list and update it, what happens? I would expect only new items to be rendered, not a whole list re-render. Is that testable?

Reply:
Ah - are you saying, test to make sure the object itself isn't replaced?